react-app/src/navigation/RouterAnalytics.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from "react";
+import * as ReactGA from "react-ga";
+import { useLocation } from "react-router-dom";
+import { useEffectOnce } from "../hooks/useEffectOnce";
+
+interface RouterAnalyticsProps {
+  id: string | null;
+  children: React.ReactNode;
+}
+
+const RouterAnalytics: React.FC<RouterAnalyticsProps> = (props) => {
+  const { id, children } = props;
+  const location = useLocation();
+  const [isInitialized, setIsInitialized] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!isInitialized) return;
+
+    ReactGA.pageview(location.pathname + location.search);
+  }, [location, isInitialized]);
+
+  useEffectOnce(
```

Comment:
Let's use useEffect with empty dep array if we are not waiting for first occurrence of an condition